system-test/system.ts

```diff
@@ -5,4 +5,122 @@
  * See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
  */
 
-console.warn(`no system tests available 👻`);
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as gcx from 'gcx';
+import {cloudfunctions_v1, google} from 'googleapis';
+import fetch from 'node-fetch';
+import * as path from 'path';
+import {promisify} from 'util';
+import * as uuid from 'uuid';
+
+const spawn = require('await-spawn');
+const mv = promisify(fs.rename);
+const pkg = require('../../package.json');
+const projectId = process.env.GCLOUD_PROJECT;
+
+if (!projectId) {
+  throw new Error('Please set the `GCLOUD_PROJECT` environment variable.');
+}
+
+let gcf: cloudfunctions_v1.Cloudfunctions;
+const shortPrefix = 'gcloud-tests';
+const fullPrefix = `${shortPrefix}-${uuid.v4().split('-')[0]}`;
+
+describe('gcp metadata', () => {
+  before(async () => {
+    // Clean up any old cloud functions just hanging out
+    gcf = await getGCFClient();
+    await pruneFunctions(false);
```

Comment:
AFAICT, this library doesn't terminate a test on an unhandled rejection. Can you add a `mocha.opts` with `throwDeprecation`.

Reply:
So this is super weird, but `mocha` respects the `mocha.opts` in the `test` directory, even if you're running tests in a *different* directory :)  In there, we `--throw-deprecation` so we're safe. I tested this by throwing and making sure it exited with a typescript stack trace (it's using source-map-support).